### PR TITLE
feat: value prop added for prefered Ingredients, fixed buttons (fixed branch)

### LINKED ISSUE #64: PreferredIngredients component should get a value prop to render the current user profile data

### PR DESCRIPTION
**Pull request by Bhoft**
fixes #64

### CODE REVIEW

File: components/preferredIngredients/preferredIngredients.tsx [hunk 1]
```diff
@@ -2,22 +2,27 @@ import { useRef, useState } from 'react'
...
   const inputRef = useRef<HTMLInputElement>(null)
 
   const handleIngredientRemove = (ingredientToRemove: string) => {
-    const selectedIngredients = ingredients.filter(
-      (ingredient) => ingredient !== ingredientToRemove,
-    )
-    setIngredients(selectedIngredients)
-    onChange(selectedIngredients)
+    // remove the ingredient when found
+    if(ingredients.find((ingredient) => ingredientToRemove === ingredient)) {
```
Comment (moncapitaine) [inline components/preferredIngredients/preferredIngredients.tsx:19]: remove, not needed

File: components/preferredIngredients/preferredIngredients.tsx [hunk 2]
```diff
@@ -2,22 +2,27 @@ import { useRef, useState } from 'react'
...
+  onChange: (selectedIngredients: string[]) => void,
+  selectedIngredients: string[]
 }
 
 export const PreferredIngredients: React.FC<PreferredIngredientsProps> = ({
   onChange,
+  selectedIngredients
 }) => {
-  const [ingredients, setIngredients] = useState<string[]>([])
+  const [ingredients, setIngredients] = useState<string[]>(selectedIngredients)
```
Comment (moncapitaine) [inline components/preferredIngredients/preferredIngredients.tsx:13]: remove, it is now a prop

File: context/UserProfileContext.ts [hunk 3]
```diff
@@ -8,7 +8,10 @@ export interface UserProfileContextProps {
 }
 
 const defaultContextProps: UserProfileContextProps = {
-  userProfile: {},
+  userProfile: {
```
Comment (moncapitaine) [inline context/UserProfileContext.ts:11]: use UserProfile interface

**Comment by moncapitaine** [comment]
please change my remarks

File: components/preferredIngredients/preferredIngredients.tsx (hunk 2 above)
Comment (Bhoft) [reply components/preferredIngredients/preferredIngredients.tsx]: done

File: components/preferredIngredients/preferredIngredients.tsx (hunk 1 above)
Comment (Bhoft) [reply components/preferredIngredients/preferredIngredients.tsx]: done

**Review by moncapitaine — APPROVED** [review]

File: components/preferredIngredients/preferredIngredients.tsx [hunk 4]
```diff
@@ -56,12 +61,14 @@ export const PreferredIngredients: React.FC<PreferredIngredientsProps> = ({
           type="text"
           ref={inputRef}
         />
-        <button
-          className="rounded-full border p-2 hover:bg-green-300 bg-green-100"
-          onClick={handleIngredientAdd}
-        >
-          <GrAdd aria-label="add" />
-        </button>
+        <div>
```
[the remark this answers is not in the document]
Comment (Bhoft) [reply components/preferredIngredients/preferredIngredients.tsx:57]: left in :)

**Review by pSchwietzer — APPROVED** [review]

**Review by DexHarris — APPROVED** [review]

**Review by taylorichigo — APPROVED** [review]

**Pull request merged**